pkg/remote/remote_catalog.go

```diff
@@ -130,7 +133,8 @@ func (c *RemoteCatalog) createOrUpdateCatalog(ctx context.Context,
 
 	// it can happen that the ModuleTemplate CRD is not existing in the Remote Cluster when we apply it and retry
 	if containsMetaIsNoMatchErr(errs) {
-		if err := c.CreateModuleTemplateCRDInRuntime(ctx, v1beta2.ModuleTemplateKind.Plural()); err != nil {
+		if err := c.CreateModuleTemplateCRDInRuntime(ctx, v1beta2.ModuleTemplateKind.Plural(),
```

Comment:
Same here, it's only called when CRD is not found, we don't need cache

Reply:
Same here, the cache is only for the KCP so it doesn't guarantee that it's there for the runtime